packages/shared/src/components/squads/PublicSquadEligibilityCard.tsx

```diff
@@ -10,42 +16,46 @@ import { useLazyModal } from '../../hooks/useLazyModal';
 import { LazyModal } from '../modals/common/types';
 import { anchorDefaultRel } from '../../lib/strings';
 import {
-  SquadPostsProgressBar,
   MIN_SQUAD_POSTS,
+  SquadPostsProgressBar,
 } from './SquadPostsProgressBar';
 import { squadsPublicGuide } from '../../lib/constants';
+import { SquadStatus } from './settings';
 
-interface Props {
-  postsCount: number;
-  inReview: boolean;
-  squadId: string;
+interface PublicSquadEligibilityCardProps {
+  placeholder: ReactNode;
 }
 
-const PublicSquadEligibilityCard = ({
-  postsCount,
-  inReview,
-  squadId,
-}: Props): ReactElement => {
+export function PublicSquadEligibilityCard({
+  placeholder,
+}: PublicSquadEligibilityCardProps): ReactElement {
+  const router = useRouter();
+  const { squad } = useSquad({ handle: router.query.handle as string });
+  const { isFetched, status } = usePublicSquadRequests({
+    sourceId: squad?.id,
+  });
+  const postsCount = squad?.flags.totalPosts;
```

Comment:
Is there a possibility where server side props have squad, but not fetched flags yet?
Maybe make it `?` as well just to be safe?

Reply:
Yeah, sure, we can add that.